client/network/src/service.rs

```diff
@@ -217,12 +212,8 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
 
 		// Initialize the reserved peers.
 		for reserved in params.network_config.reserved_nodes.iter() {
-			if let Ok((peer_id, addr)) = parse_str_addr(reserved) {
-				reserved_nodes.push(peer_id.clone());
-				known_addresses.push((peer_id, addr));
-			} else {
-				warn!(target: "sub-libp2p", "Not a valid reserved node address: {}", reserved);
-			}
+			reserved_nodes.push(reserved.peer_id.clone());
+			known_addresses.push((reserved.peer_id.clone(), reserved.multiaddr.clone()));
 		}
```

Comment:
We should add nodes from `Role::Sentry { validators }` or `Role::Authority { sentry_nodes }` to some peerset priority group.